builtin/providers/aws/resource_aws_ecs_task_definition.go

```diff
@@ -132,18 +132,34 @@ func resourceAwsEcsTaskDefinitionRead(d *schema.ResourceData, meta interface{})
 }
 
 func resourceAwsEcsTaskDefinitionUpdate(d *schema.ResourceData, meta interface{}) error {
-	return resourceAwsEcsTaskDefinitionCreate(d, meta)
+	oldArn := d.Get("arn").(string)
+
+	log.Printf("[DEBUG] Creating new revision of task definition %q", d.Id())
+	err := resourceAwsEcsTaskDefinitionCreate(d, meta)
+	if err != nil {
+		return err
+	}
+	log.Printf("[DEBUG] New revision of %q created: %q", d.Id(), d.Get("arn").(string))
+
+	log.Printf("[DEBUG] Unregistering old revision of task definition %q: %q", d.Id(), oldArn)
+	conn := meta.(*AWSClient).ecsconn
+	_, err = conn.DeregisterTaskDefinition(&ecs.DeregisterTaskDefinitionInput{
+		TaskDefinition: aws.String(oldArn),
+	})
+	return err
 }
```

Comment:
we often return the results of the `read` method when we're finished in the `update` method, is that appropriate here?